sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_first_seen_v1/query.sql

```diff
@@ -0,0 +1,40 @@
+WITH _current AS (
+  SELECT
+    user_id,
+    ARRAY_AGG(DISTINCT service) AS services_used,
+  FROM
+    firefox_accounts.fxa_all_events
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+  GROUP BY
+    user_id
+),
+_previous AS (
+  SELECT
+    *
+  FROM
+    firefox_accounts_derived.fxa_users_first_seen_v1
+  WHERE
+    first_seen_date < @submission_date
```

Comment:
```suggestion
    -- If reprocessing scenarios, we must always backfill from the first affected date
    -- all the way to the present; to enforce that, we explicitly drop any data after
    -- the target @submission_date
    first_seen_date < @submission_date
```

This is surprising enough that I think it warrants commentary.